File: Emulator/src/main/content/global/skill/summoning/familiar/npc/PrayingMantisNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.Familiar;
import content.global.skill.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;

/**
 * The type Praying mantis npc.
 */
@Initializable
public class PrayingMantisNPC extends content.global.skill.summoning.familiar.Familiar {

    /**
     * Instantiates a new Praying mantis npc.
     */
    public PrayingMantisNPC() {
        this(null, 6798);
    }

    /**
     * Instantiates a new Praying mantis npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public PrayingMantisNPC(Player owner, int id) {
        super(owner, id, 6900, 12011, 6, WeaponInterface.STYLE_ACCURATE);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new PrayingMantisNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        return false;
    }

    @Override
    public int[] getIds() {
        return new int[]{6798, 6799};
    }

}

File: Emulator/src/main/content/global/skill/summoning/familiar/npc/PrayingMantisNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.Familiar
import content.global.skill.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs.consts.Items
import org.rs.consts.NPCs

/**
 * The type Praying mantis npc.
 */
@Initializable
class PrayingMantisNPC
/**
 * Instantiates a new Praying mantis npc.
 *
 * @param owner the owner
 * @param id    the id
 */
/**
 * Instantiates a new Praying mantis npc.
 */
@JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.PRAYING_MANTIS_6798) :
    Familiar(owner, id, 6900, Items.PRAYING_MANTIS_POUCH_12011, 6, WeaponInterface.STYLE_ACCURATE) {
    override fun construct(owner: Player, id: Int): Familiar {
        return PrayingMantisNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        return false
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.PRAYING_MANTIS_6798, NPCs.PRAYING_MANTIS_6799)
    }
}
